trin-bridge/src/bridge.rs

```diff
@@ -200,23 +217,34 @@ impl Bridge {
     }
 
     async fn serve_full_block(
+        &self,
         height: u64,
         epoch_acc: Option<Arc<EpochAccumulator>>,
         gossip_stats: Arc<Mutex<GossipStats>>,
         portal_clients: Vec<HttpClient>,
     ) -> anyhow::Result<()> {
         debug!("Serving block: {height}");
-        let mut full_header = Bridge::get_header(height).await?;
+        let mut full_header = self.pandaops.get_header(height).await?;
         if full_header.header.number <= MERGE_BLOCK_NUMBER {
             full_header.epoch_acc = epoch_acc;
         }
         Bridge::gossip_header(&full_header, &portal_clients, &gossip_stats).await?;
         // Sleep for 10 seconds to allow headers to saturate network,
         // since they must be available for body / receipt validation.
         sleep(Duration::from_secs(HEADER_SATURATION_DELAY)).await;
-        Bridge::construct_and_gossip_block_body(&full_header, &portal_clients, &gossip_stats)
-            .await?;
-        Bridge::construct_and_gossip_receipt(&full_header, &portal_clients, &gossip_stats).await
+        if let Err(err) = self
```

Comment:
Curious why you changed to log the errors at this level? Imo, it's cleaner to log it at the top-level (aka inside the `launch_*` methods).
1. It removes the `let _ = ...` clause from the `launch_*` methods which is messy imo
2. In this case I think it's cleaner to let the error propagate all the way up and log them there.

If you're looking for some enhanced granularity in the log msg, maybe we could just `map_err()` at this level to add context to the error message...

```rust
        Bridge::construct_and_gossip_block_body(&full_header, &portal_clients, &gossip_stats)
            .await
            .map_err(|err| anyhow!("Error serving block body: {err:?}"))?;
```